File: src/main/java/com/github/awruff/rawinput/RawInputMod.java
package com.github.awruff.rawinput;

import com.github.awruff.rawinput.impl.RawMouseInput;
import com.github.awruff.rawinput.utils.LibraryChecker;
import net.minecraft.client.Minecraft;
import org.apache.commons.lang3.SystemUtils;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class RawInputMod {
    public final static String ID = "rawinput";
    public final static String NAME = "Raw Input";
    public final static String VERSION = "1.0.0";

    private static final Logger LOGGER = LogManager.getLogger(NAME);

    public static String environment = "";

    public static void initialize() {
        LOGGER.debug("Initializing");
        boolean direct = LibraryChecker.isLibraryLoaded("jinput-dx8");
        boolean raw = LibraryChecker.isLibraryLoaded("jinput-raw");

        if (raw || direct) {
            if (raw) {
                environment = "DirectAndRawInputEnvironmentPlugin";
            } else {
                environment = "DirectInputEnvironmentPlugin";
            }

            LOGGER.debug("Environment set to {}", environment);

            Minecraft.getInstance().mouse = new RawMouseInput();
        }

        if (environment.isEmpty()) {
            LOGGER.warn("Environment is blank... {} {} {}", SystemUtils.OS_NAME, SystemUtils.OS_VERSION, SystemUtils.OS_ARCH);
        }
    }
}

File: src/main/kotlin/com/github/awruff/rawinput/RawInputMod.kt
package com.github.awruff.rawinput

import com.github.awruff.rawinput.impl.RawMouseInput
import com.github.awruff.rawinput.utils.LibraryChecker
import net.minecraft.client.Minecraft
import org.apache.commons.lang3.SystemUtils
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

object RawInputMod {
    const val ID = "rawinput"
    const val NAME = "Raw Input"
    const val VERSION = "1.1.0"

    private val LOGGER: Logger = LogManager.getLogger(NAME)

    private const val RAW_PLUGIN = "DirectAndRawInputEnvironmentPlugin"
    private const val DX8_PLUGIN = "DirectInputEnvironmentPlugin"

    val environment: String by lazy {
        when {
            LibraryChecker.isLibraryPresent("jinput-raw") -> RAW_PLUGIN
            LibraryChecker.isLibraryPresent("jinput-dx8") -> DX8_PLUGIN
            else -> ""
        }
    }

    fun initialize() {
        LOGGER.info("Initializing {} v{} (id={})", NAME, VERSION, ID)

        if (environment.isNotEmpty()) {
            LOGGER.info("Using input environment: {}", environment)
            Minecraft.getInstance().mouse = RawMouseInput
        } else {
            LOGGER.warn(
                "No suitable input environment found. Checked libraries: [jinput-raw, jinput-dx8]. " +
                        "System details: OS={} Version={} Arch={}",
                SystemUtils.OS_NAME,
                SystemUtils.OS_VERSION,
                SystemUtils.OS_ARCH
            )
        }
    }
}